src/lib/set-controller.ts

```diff
@@ -10,12 +15,14 @@ import { checkNetwork } from "./check-network.js";
 export async function setController(
   this: Connection,
   controller: string,
-  name: string
+  name: string,
+  options?: SetControllerOptions
 ): Promise<WriteQueryResult> {
   const tableId = name.trim().split("_").pop();
   if (typeof tableId !== "string") throw new Error("malformed tablename");
 
-  if (this.options.rpcRelay) {
+  const doRelay = shouldRelay(this, options);
+  if (doRelay) {
```

Comment:
This allows the caller to specify if they want to use the relay on a per call basis.